src/seedu/addressbook/AddressBook.java

```diff
@@ -383,12 +393,28 @@ private static String executeCommand(String userInputString) {
             return getUsageInfoForAllCommands();
         case COMMAND_EXIT_WORD:
             executeExitProgramRequest();
+        case COMMAND_SORT_WORD:
+        	return executeSortAllPersonsInAddressBook();
         default:
-            return getMessageForInvalidCommandInput(commandType, getUsageInfoForAllCommands());
+            return getMessageForInvalidCommandInput(
+            		commandType, getUsageInfoForAllCommands());
         }
-    }
-
-    /**
+	}
+
+    private static String executeSortAllPersonsInAddressBook() {
+        ArrayList<HashMap<PersonProperty, String>> toBeDisplayed = getAllPersonsInAddressBook();
+        ArrayList<HashMap<PersonProperty, String>> personsCopy = new ArrayList<>(toBeDisplayed);
+        Collections.sort(personsCopy, new Comparator<HashMap<PersonProperty, String>>(){
+        	@Override
+        	public int compare(HashMap<PersonProperty, String> personA, HashMap<PersonProperty, String> personB){
+        		return personA.get(PersonProperty.NAME).compareTo(personB.get(PersonProperty.NAME));
+        	}
+        });
+        showToUser(personsCopy);
+        return getMessageForPersonsDisplayedSummary(personsCopy);
+	}
+    
+	/**
```

Comment:
Great attempt in making full use of the Java API and devising the comparator.  Just need to bear in mind code quality though...